tests/test_preprocess_artifacts.py

```diff
@@ -2,72 +2,75 @@
 
 import inspect
 import os
+import pathlib
+
 import pytest
 
 from .utils import LocalTestEnvironment
+from .conftest import FuzzyCallChecker
+
+
+class ArtifactsTestEnvironment(LocalTestEnvironment):
+
+    def __init__(self, tmpdir: pathlib.Path):
+        super().__init__(tmpdir, "main")
+        self.artifact_name: str = "artifact"
+        self.artifact_path: pathlib.Path = self.artifact_dir.join(self.artifact_name)
+        self.artifact_content: str = "artifact content"
+
+    def write_config_file(self, artifact_prebuild_clean: bool):
+        config: str = inspect.cleandoc(f"""
+            from universum.configuration_support import Configuration, Step
+            step = Step(name='Step',
+                        command=['bash', '-c', 'echo "{self.artifact_content}" > {self.artifact_name}'],
+                        artifacts='{self.artifact_name}',
+                        artifact_prebuild_clean={artifact_prebuild_clean})
+            configs = Configuration([step])
+        """)
+        self.configs_file.write_text(config, "utf-8")
+
+    def check_step_artifact_present(self):
+        assert os.path.exists(self.artifact_path)
+        with open(self.artifact_path, encoding="utf-8") as f:
+            content: str = f.read().replace("\n", "")
```

Comment:
note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs